### PR TITLE
multiple refactors and fixes

### PR DESCRIPTION
**Pull request by vieux**
sorry it's a big PR, reading commit by commit could be helpful.
I started refactoring one side and then `¯\_(ツ)_/¯` started from https://github.com/docker/go-plugins-helpers/pull/76

The bunch of the refactor started from here.

some API (graphdriver) had a `Err` field in almost every Responses, and sometimes duplicated (empty) responses
some API (volume) had a `Err` field in *some* Responses and sometimes used `ErrorResponse`.
some API (network, ipam) had no `Err` and used a dedicated `ErrorResponse` for error.

Basically graphdriver and network/ipam were totally different and volume was in the middle.
I decided to uniform everything like network and ipam.

It also adds a bunch of fixes, such as adding missing multiple `return`

ping @cpuguy83 for some reason I cannot add you a reviewer.

### CODE REVIEW

File: ipam/api.go [hunk 1]
```diff
@@ -170,9 +165,9 @@ func (h *Handler) initMux() {
 		}
 		err = h.ipam.ReleaseAddress(req)
 		if err != nil {
-			msg := err.Error()
-			sdk.EncodeResponse(w, NewErrorResponse(msg), msg)
+			sdk.EncodeResponse(w, NewErrorResponse(err.Error()), true)
+			return
```
Comment (vieux) [inline ipam/api.go]: missing return was causing 2 responses being sent.

File: network/api.go [hunk 2]
```diff
@@ -328,10 +320,10 @@ func (h *Handler) initMux() {
 		}
 		res, err := h.driver.EndpointInfo(req)
 		if err != nil {
-			msg := err.Error()
-			sdk.EncodeResponse(w, NewErrorResponse(msg), msg)
+			sdk.EncodeResponse(w, NewErrorResponse(err.Error()), true)
+			return
```
Comment (vieux) [inline network/api.go]: missing return was causing 2 responses being sent.

File: sdk/encoder.go [hunk 3]
```diff
@@ -19,9 +19,9 @@ func DecodeRequest(w http.ResponseWriter, r *http.Request, req interface{}) (err
 }
 
 // EncodeResponse encodes the given structure into an http response.
-func EncodeResponse(w http.ResponseWriter, res interface{}, err string) {
+func EncodeResponse(w http.ResponseWriter, res interface{}, err bool) {
```
Comment (vieux) [inline sdk/encoder.go]: no point taking a string, it was never read.

File: volume/shim/shim.go [hunk 4]
```diff
@@ -47,7 +46,6 @@ func (d *shimDriver) Get(req *volumeplugin.GetRequest) (*volumeplugin.GetRespons
 	var res *volumeplugin.GetResponse
 	v, err := d.d.Get(req.Name)
 	if err != nil {
-		res.Err = err.Error()
```
Comment (vieux) [inline volume/shim/shim.go]: was never used

**Comment by vieux** [comment]
ping @cpuguy83 @dave-tucker <3

**Comment by dave-tucker** [comment]
LGTM 🥇 Thanks @vieux
